server/src/main/resources/templates/web/idp_discovery/account_chooser.html

```diff
@@ -20,13 +20,14 @@ <h4 class="txt-c pbxxl ptxl" th:text="${T(org.springframework.util.StringUtils).
       <li class="list-group-item" th:each="account, iter : ${savedAccounts}" th:inline="text">
         <form th:id="${account.userId}" action="/login/idp_discovery" th:action="@{/login/idp_discovery}" method="post" role="form">
           <input type="hidden" class="form-control" name="email" th:value="${account.origin == 'uaa' || account.origin == 'ldap' ? account.username : account.email}" />
+          <input type="hidden" class="form-control" name="login_hint" th:value="${'%7B%22origin%22:%22'+account.origin+'%22%7D'}" />
```

Comment:
I'm surprised this json fragment needed to be urlencoded by hand.  Won't thymeleaf handle the encoding?

Reply:
It probably would do that, but putting the double quotes there without enconding or escaping would break the statement. 
So I chose to url encode it manually, as both thymeleaf  can work with that and the UAA knows how to handle the url encoded value.